### ISSUE TITLE
Request Contrib Group membership

### ISSUE THREAD

**Issue by sentaidigital** [backdrop-ops/contrib#141]
I would like to join the Backdrop CMS developer community.  One of my sandbox modules, Date Format Help, has been ported from D7 to Backdrop 1.x and is ready for review.

The original sandbox is here: https://www.drupal.org/sandbox/franklin/1728758

The ported module is here: https://github.com/sd-backdrop/date_format_help

**Comment by klonos** [comment]
@sentaidigital Hello there John and welcome to Backdrop contrib! Would you care to move it over to the official contrib repository so that its releases are automatically available in the list of modules in backdropcms.org?

New project checklist
---------------------

All projects must meet these minimum requirements.

- [ ] Include a LICENSE.txt file that indicates the code is GPL v2. You can use [this copy](https://raw.githubusercontent.com/backdrop-ops/contrib/master/examples/LICENSE.txt).
- [ ] Include a README.md file that includes license and maintainer information. You can use [this example](https://raw.githubusercontent.com/backdrop-ops/contrib/master/examples/README.md).
- [ ] Maintain the Git history from Drupal 7. See [this article](http://tag1consulting.com/blog/how-maintain-contrib-modules-drupal-and-backdrop-same-time-part-2).

**Comment by klonos** [comment]
...I've checked and found that you already meet the two first requirements. Pending on the 3rd.

**Comment by klonos** [comment]
...commit history seems to be retained from d.org. [There are commits](https://github.com/sd-backdrop/date_format_help/commits/1.x-1.x) that date as far back as August 2012 and there is also a 6.x-1.x and a 7.x-1.x branch. So I think you are good.

For the record, do you agree to the below statement:

Backdrop Contributed Project Agreement
--------------------------------------

By joining the [Backdrop Contributed Project Group](https://github.com/backdrop-contrib) (Backdrop Contrib, for short) you agree to the following:

1. You will not push changes to a repository for which you are not a current maintainer (even though joining the Backdrop Contrib group will grant you access to push to any project within).

1. You must agree to license your code contributions as GPLv2 or later.

1. Any project you create or maintain must include a copy of the [GPL v2 `LICENSE.txt` file](https://github.com/backdrop-ops/contrib/blob/master/examples/LICENSE.txt) in the root of your repository. The GPLv2 license applies to all code that directly interacts with parts of Backdrop licensed as GPLv2 or later.  See for the [Backdrop License FAQ](https://backdropcms.org/license) for a more detailed explanation.

1. You must confirm that you have the right to distribute any additional code, libraries, images, fonts or other assets written or created by any third party with code licensed as GPLv2 or later.

1. Any project you create or maintain must include a `README.md` file containing at the least the following:
  1. A description of the project
  1. Basic documentation
  1. License information for the project (GPL v2)
  1. License information for any additional assets (SIL OFL fonts, CC-SA images, etc)
  1. A list of the current maintainers for the Backdrop project
  1. Credits acknowledging past maintainers for the Backdrop or Drupal projects

  You may use this [example README.md](https://github.com/backdrop-ops/contrib/blob/master/examples/README.md) to get started.

1. You will work with the Backdrop Security Team to address any vulnerabilities in any project you create or maintain, if necessary.

1. Any project you create or maintain will have the GitHub issue tracker enabled for official communication.

1. If any of the above requirements are not met, your access to the Backdrop Contrib group -- including all projects and even those that you may have originally authored -- may be revoked.

1. If your project becomes abandoned and you do not respond to requests in the issue tracker within 2 weeks, your project may be modified by a Backdrop Contrib administrator without your explicit consent.

---

**Comment by sentaidigital** [comment]
I consent to the Backdrop Contributed Project Agreement.

**Comment by klonos** [comment]
Great! Just one minor delay before we send you an invitation...

@quicksketch so that I'm not running ahead of myself trying to help you with this, do you need to do a code review first?

**Comment by docwilmot** [comment]
We've never required a code review first, to my knowlege. None of my crap would be in contrib if that was the case. :smile: Go ahead and let 'im in!

**Comment by klonos** [comment]
@docwilmot's reasoning always good enough for me.

@sentaidigital invitation on it's way!

**Comment by sentaidigital** [comment]
Thank you, @klonos.  Happy to be here. 

When I accepted the invitation, GitHub decided I wanted to watch EVERY SINGLE Backdrop module's repository.  Email below.

> Hey there, we’re just writing to let you know that you’ve automatically started watching several repositories on GitHub.
> 
> You’ll receive notifications for all issues, pull requests, and comments that happen inside the repository. If you would like to stop watching any of these repositories, you can manage your settings here:
> 
>  https://github.com/backdrop-contrib/devel/subscription
>  https://github.com/backdrop-contrib/simplify/subscription
>  https://github.com/backdrop-contrib/extlink/subscription
>  https://github.com/backdrop-contrib/token/subscription
>  https://github.com/backdrop-contrib/nice_menus/subscription
>  https://github.com/backdrop-contrib/nivo_slider/subscription
>  https://github.com/backdrop-contrib/cleanish/subscription
>  https://github.com/backdrop-contrib/forum/subscription
>  https://github.com/backdrop-contrib/profile_picture_plus/subscription
>  https://github.com/backdrop-contrib/codefilter/subscription
>  https://github.com/backdrop-contrib/pathauto/subscription
>  https://github.com/backdrop-contrib/email/subscription
>  https://github.com/backdrop-contrib/codemirror/subscription
>  https://github.com/backdrop-contrib/aeon5/subscription
>  https://github.com/backdrop-contrib/markdown/subscription
>  https://github.com/backdrop-contrib/captcha/subscription
>  https://github.com/backdrop-contrib/recaptcha/subscription
>  https://github.com/backdrop-contrib/api/subscription
>  https://github.com/backdrop-contrib/bueditor/subscription
>  https://github.com/backdrop-contrib/site_info_blocks/subscription
>  https://github.com/backdrop-contrib/page_background/subscription
>  https://github.com/backdrop-contrib/imce/subscription
>  https://github.com/backdrop-contrib/insert/subscription
>  https://github.com/backdrop-contrib/radix/subscription
>  https://github.com/backdrop-contrib/markdowneditor/subscription
>  https://github.com/backdrop-contrib/ajax_markup/subscription
>  https://github.com/backdrop-contrib/google_analytics/subscription
>  https://github.com/backdrop-contrib/backup_migrate/subscription
>  https://github.com/backdrop-contrib/user_cancel_password_confirm/subscription
>  https://github.com/backdrop-contrib/redirect/subscription
>  https://github.com/backdrop-contrib/oauth/subscription
>  https://github.com/backdrop-contrib/special_menu_items/subscription
>  https://github.com/backdrop-contrib/custom_search/subscription
>  https://github.com/backdrop-contrib/wormhole/subscription
>  https://github.com/backdrop-contrib/custom_header_block/subscription
>  https://github.com/backdrop-contrib/aggregator/subscription
>  https://github.com/backdrop-contrib/on_the_web/subscription
>  https://github.com/backdrop-contrib/insert_view/subscription
>  https://github.com/backdrop-contrib/comment_notify/subscription
>  https://github.com/backdrop-contrib/video_filter/subscription
>  https://github.com/backdrop-contrib/link/subscription
>  https://github.com/backdrop-contrib/radix_layouts/subscription
>  https://github.com/backdrop-contrib/formblock/subscription
>  https://github.com/backdrop-contrib/googleanalytics/subscription
>  https://github.com/backdrop-contrib/examples/subscription
>  https://github.com/backdrop-contrib/twitter/subscription
>  https://github.com/backdrop-contrib/compact_forms/subscription
>  https://github.com/backdrop-contrib/menu_trail_by_path/subscription
>  https://github.com/backdrop-contrib/ayah/subscription
>  https://github.com/backdrop-contrib/search_config/subscription
>  https://github.com/backdrop-contrib/ckeditor/subscription
>  https://github.com/backdrop-contrib/honeypot/subscription
>  https://github.com/backdrop-contrib/custom_meta/subscription
>  https://github.com/backdrop-contrib/rua/subscription
>  https://github.com/backdrop-contrib/pace/subscription
>  https://github.com/backdrop-contrib/auto_nodetitle/subscription
>  https://github.com/backdrop-contrib/basic_cart/subscription
>  https://github.com/backdrop-contrib/simple_paypal_donate/subscription
>  https://github.com/backdrop-contrib/instagram_block/subscription
>  https://github.com/backdrop-contrib/colorbox/subscription
>  https://github.com/backdrop-contrib/sharethis/subscription
>  https://github.com/backdrop-contrib/disqus/subscription
>  https://github.com/backdrop-contrib/node_export/subscription
>  https://github.com/backdrop-contrib/uuid/subscription
>  https://github.com/backdrop-contrib/references/subscription
>  https://github.com/backdrop-contrib/fanciblock/subscription
>  https://github.com/backdrop-contrib/webform/subscription
>  https://github.com/backdrop-contrib/transliteration/subscription
>  https://github.com/backdrop-contrib/date/subscription
>  https://github.com/backdrop-contrib/rules/subscription
>  https://github.com/backdrop-contrib/login_destination/subscription
>  https://github.com/backdrop-contrib/menu_position/subscription
>  https://github.com/backdrop-contrib/flag/subscription
>  https://github.com/backdrop-contrib/submitted_by/subscription
>  https://github.com/backdrop-contrib/logintoboggan/subscription
>  https://github.com/backdrop-contrib/float_filter/subscription
>  https://github.com/backdrop-contrib/contact_forms/subscription
>  https://github.com/backdrop-contrib/views_slideshow/subscription
>  https://github.com/backdrop-contrib/libraries/subscription
>  https://github.com/backdrop-contrib/addanother/subscription
>  https://github.com/backdrop-contrib/simplenews_scheduler/subscription
>  https://github.com/backdrop-contrib/mailsystem/subscription
>  https://github.com/backdrop-contrib/mandrill/subscription
>  https://github.com/backdrop-contrib/mimemail/subscription
>  https://github.com/backdrop-contrib/simplenews/subscription
>  https://github.com/backdrop-contrib/smtp/subscription
>  https://github.com/backdrop-contrib/webform_remote_post/subscription
>  https://github.com/backdrop-contrib/globalredirect/subscription
>  https://github.com/backdrop-contrib/views_load_more/subscription
>  https://github.com/backdrop-contrib/access_unpublished/subscription
>  https://github.com/backdrop-contrib/waypoints/subscription
>  https://github.com/backdrop-contrib/void_menu/subscription
>  https://github.com/backdrop-contrib/scrollreveal/subscription
>  https://github.com/backdrop-contrib/parallax_bg/subscription
>  https://github.com/backdrop-contrib/joyride/subscription
>  https://github.com/backdrop-contrib/apc/subscription
>  https://github.com/backdrop-contrib/pathologic/subscription
>  https://github.com/backdrop-contrib/commentsblock/subscription
>  https://github.com/backdrop-contrib/hopscotch/subscription
>  https://github.com/backdrop-contrib/efq_helper/subscription
>  https://github.com/backdrop-contrib/two_column_flexible/subscription
>  https://github.com/backdrop-contrib/sticky_admin_tabs/subscription
>  https://github.com/backdrop-contrib/project_browser/subscription
>  https://github.com/backdrop-contrib/project_browser_server/subscription
>  https://github.com/backdrop-contrib/mobile_navigation/subscription
>  https://github.com/backdrop-contrib/materialize/subscription
>  https://github.com/backdrop-contrib/news_arrow/subscription
>  https://github.com/backdrop-contrib/summer_fun/subscription
>  https://github.com/backdrop-contrib/zurb_foundation_5/subscription
>  https://github.com/backdrop-contrib/user_disk_quota/subscription
>  https://github.com/backdrop-contrib/autofloat/subscription
>  https://github.com/backdrop-contrib/flickr/subscription
>  https://github.com/backdrop-contrib/auto_entitylabel/subscription
>  https://github.com/backdrop-contrib/filefield_sources/subscription
>  https://github.com/backdrop-contrib/google_fonts/subscription
>  https://github.com/backdrop-contrib/fences/subscription
>  https://github.com/backdrop-contrib/youtube/subscription
>  https://github.com/backdrop-contrib/senjo/subscription
>  https://github.com/backdrop-contrib/colihaut/subscription
>  https://github.com/backdrop-contrib/jquery_expander/subscription
>  https://github.com/backdrop-contrib/mero/subscription
>  https://github.com/backdrop-contrib/invisimail/subscription
>  https://github.com/backdrop-contrib/smsframework/subscription
>  https://github.com/backdrop-contrib/sms_clickatell/subscription
>  https://github.com/backdrop-contrib/date_basic/subscription
>  https://github.com/backdrop-contrib/views_send/subscription
>  https://github.com/backdrop-contrib/form_wizard/subscription
>  https://github.com/backdrop-contrib/setup/subscription
>  https://github.com/backdrop-contrib/contest/subscription
>  https://github.com/backdrop-contrib/blockify/subscription
>  https://github.com/backdrop-contrib/feed_import/subscription
>  https://github.com/backdrop-contrib/style_settings/subscription
>  https://github.com/backdrop-contrib/back_to_top/subscription
>  https://github.com/backdrop-contrib/job_scheduler/subscription
>  https://github.com/backdrop-contrib/responsive_menus/subscription
>  https://github.com/backdrop-contrib/afterlight_tribute/subscription
>  https://github.com/backdrop-contrib/salvattore/subscription
>  https://github.com/backdrop-contrib/video_embed_field/subscription
>  https://github.com/backdrop-contrib/date_all_day/subscription
>  https://github.com/backdrop-contrib/date_repeat/subscription
>  https://github.com/backdrop-contrib/date_tools/subscription
>  https://github.com/backdrop-contrib/elements/subscription
>  https://github.com/backdrop-contrib/flexviews/subscription
>  https://github.com/backdrop-contrib/font_awesome/subscription
>  https://github.com/backdrop-contrib/wpmenu/subscription
>  https://github.com/backdrop-contrib/entity_blocks/subscription
>  https://github.com/backdrop-contrib/custom_text_block/subscription
>  https://github.com/backdrop-contrib/b/subscription
>  https://github.com/backdrop-contrib/bootstrap_lite/subscription
>  https://github.com/backdrop-contrib/base_meta/subscription
>  https://github.com/backdrop-contrib/gcal/subscription
>  https://github.com/backdrop-contrib/agenda/subscription
>  https://github.com/backdrop-contrib/drush/subscription
>  https://github.com/backdrop-contrib/simple_access/subscription
>  https://github.com/backdrop-contrib/eu-cookie-compliance/subscription
>  https://github.com/backdrop-contrib/form-element-layout/subscription
>  https://github.com/backdrop-contrib/spam_bot_blocker/subscription
>  https://github.com/backdrop-contrib/tipsy/subscription
>  https://github.com/backdrop-contrib/calendar/subscription
>  https://github.com/backdrop-contrib/unsubscribe/subscription
>  https://github.com/backdrop-contrib/page_theme/subscription
>  https://github.com/backdrop-contrib/content_theme/subscription
>  https://github.com/backdrop-contrib/copyright_notice/subscription
> 
> You automatically watched these repositories because you’ve been given push access to them.
> 
> Thanks!

Um..... thanks, GitHub, but I really don't need notifications on ALL of them.  Is there some way to disable this default setting?

**Comment by gifad** [comment]
@sentaidigital : https://github.com/watching

**Comment by klonos** [comment]
Sorry about that @sentaidigital I didn't see any related option when I was adding you to the contrib members team. If I understand correctly from this:

> You automatically watched these repositories because you’ve been given push access to them.

...this is the default thing and it comes with being a member of this group. I understand that if this is the case though, it becomes a bit of an annoyance for new members (this is a list of 163 projects - imagine what will happen if there were thousands of them though). Someone with more GitHub experience should shed some light here.

In case there was an option for this and I missed it some how, terribly sorry.

**Comment by klonos** [comment]
...ahh yes, there is a "magic" unwatch all button in the link that @gifad posted, but **be careful** because that will unwatch **ALL** GitHub repos you are watching - not just backdrop contrib.

**Comment by sentaidigital** [comment]
@klonos, Drupal has 17k modules listed, of which about 2,200 are used by more than 1,000 sites.  Backdrop currently has a small fraction of those modules, but I think 2,200 is a reasonable number to plan around.  As you say, when there is a thousand modules and a long weekend (think Thanksgiving in the US or Christmas across Europe and the Americas), our poor inboxes will be buried.

If we don't already have a [DX] issue asking GitHub to add an "unwatch all in group X" button on the same link @gifad posted and/or on the group home page, that would be a good start.

**Comment by sentaidigital** [comment]
Another notification setting: https://github.com/settings/notifications  A checkbox (on by default) enables automatic watching of all repositories with push access.

**Comment by quicksketch** [comment]
> Another notification setting: https://github.com/settings/notifications A checkbox (on by default) enables automatic watching of all repositories with push access. A per-group setting would be nice.

Yeah I have this option turned off as well. My consulting work requires me to regularly get access to lots of different GitHub repositories for my clients, but after the project is done, I don't want to get notified of every commit and issue.

More recently the GitHub ability for managing permissions was significantly improved. It used to be that you had to set up a group in order to set permissions. Now you can invite and grant access to an individual on a project without a group. That gives us new ways of setting up contrib access that we did not have before. Let's talk about this separately. I made a new issue at https://github.com/backdrop-ops/contrib/issues/150.
